test/kv_test.go

```diff
@@ -106,6 +106,51 @@ func TestKeyValueBasics(t *testing.T) {
 	}
 }
 
+func TestKeyValueCrossAccount(t *testing.T) {
+	s, _ := RunServerWithConfig("configs/kv_accounts.conf")
+	defer shutdown(s)
+
+	setupBucket := func(t *testing.T) {
+		nc, err := nats.Connect(s.ClientURL(), nats.UserInfo("a", "a"))
+		expectOk(t, err)
+		defer nc.Close()
+
+		js, err := nc.JetStream()
+		expectOk(t, err)
+
+		js.DeleteKeyValue("CROSS")
+
+		_, err = js.CreateKeyValue(&nats.KeyValueConfig{Bucket: "CROSS", History: 10})
+		expectOk(t, err)
+	}
+
+	testAccount := func(t *testing.T, u, p, apiPre, kvPre string) {
+		setupBucket(t)
+
+		nc, err := nats.Connect(s.ClientURL(), nats.UserInfo(u, p))
+		expectOk(t, err)
+		defer nc.Close()
+
+		js, err := nc.JetStream(nats.APIPrefix(apiPre))
+		expectOk(t, err)
+
+		kv, err := js.KeyValue("CROSS", nats.BucketSubjectPrefix(kvPre))
+		expectOk(t, err)
+
+		_, err = kv.PutString("FOO", "BAR")
+		expectOk(t, err)
+		v, err := kv.Get("FOO")
+		expectOk(t, err)
+		if string(v.Value()) != "BAR" {
+			t.Fatalf("received wrong value: %q", v.Value())
+		}
+	}
+
+	testAccount(t, "i", "i", "fromA", "")
+	testAccount(t, "o", "o", "fromA", "X.$KV.CROSS")
+	testAccount(t, "o", "o", "fromA", "X.$KV.CROSS.")
```

Comment:
For the purpose of an arbitrary bucket prefix - thats exactly what it is, an arbitrary bucket prefix.  Its for accessing a single bucket.

As you can see from the example configuration, users can put in whatever they like we have no conventions, this supports that while we build out such conventions.

Reply:
Imagine you have 100 buckets, you export 1 of those buckes to me.  I have to import that 1 bucket, thats it.  I can import them on any arbitrary subject I choose - because we have no conventions.  So it has to import a single subject, I figured its better to support being explicit in that case.  It's the most supportable thing we can do while we figure out conventions for cross account access

Reply:
maybe it's just confusing seeing a kv like subject as the example anything prefix. Is there another example config that would make the prefix something like `bucketsFromA` or 'kvsFromA`

Reply:
well, its arbitrary you can put whatever you like there.  I updated the example a bit to make it clearer that there is no relation to names etc

Reply:
perfect. LGTM

Reply:
That's clearer, thanks.

Reply:
Does this replace the use of the inbox prefix of `forB.` which is what was used with this config? https://github.com/nats-io/nats.deno/pull/239

Reply:
No that’s needed - and this is probably why I couldn’t get watch to work. 

But that’s not really about this PR, clearly this is all too complex I don’t think any normal user can get this going. So I am ok to leave as is for now 

later when we work on adr 19 we will see about macros to simplify this

Reply:
ok yeah, I'm all for just putting something out there and revisiting it later when this is all nailed down.